File: 1423_maximumPointsYouCanObtainFromCards/main.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        frontToEnd = [0] * (k + 1)
        endToFront = [0] * (k + 1)
        for i in range(k):
            frontToEnd[i+1] = frontToEnd[i] + cardPoints[i]
            endToFront[i+1] = endToFront[i] + cardPoints[len(cardPoints)-i-1]
        
        res = 0
        for i in range(k+1):
            cur = frontToEnd[i] + endToFront[k-i]
            if cur > res:
                res = cur

        return res


    # sliding window technique
    def maxScore1(delf, cardPoints: List[int], k:int) -> int:
        leftPtr, rightPtr = 0, len(cardPoints) - k
        total = sum(cardPoints[rightPtr:])
        res = total

        while rightPtr < len(cardPoints):
            total += (cardPoints[leftPtr] - cardPoints[rightPtr])
            res = max(res, total)
            leftPtr += 1
            rightPtr += 1
        return res

File: 1423_maximumPointsYouCanObtainFromCards/main.cpp
#include <vector>

class Solution {
public:
    int maxScore(std::vector<int>& cardPoints, int k) {
        int leftPtr = 0;
        int rightPtr = cardPoints.size() - k;
        int total = 0;
        for(int i = rightPtr; i < cardPoints.size(); ++i)
            total = total + cardPoints[i];
        
        int res = total;
        for(int i = leftPtr, j = rightPtr; j < cardPoints.size(); ++i, ++j) {
            total = total + cardPoints[i] - cardPoints[j];
            if(total > res)
                res = total;
        }
        return res;
    }
};